tests/transformation_tests/test_transformation_utils.py

```diff
@@ -99,3 +101,32 @@ def test_excel_datetime_conversion():
         Timestamp("2020-09-01 00:00:00"),
         NaT,
     ]
+
+
+def test_create_dataframe():
+    data = {
+        "A": pd.Series([1, 2, 3], index=[0, 1, 2]),
+        "B": pd.Series([4, 5, 6], index=[3, 4, 5]),
+    }
+
+    # Create a DataFrame from the dictionary using pd.DataFrame constructor directly to observe problem
+    result = pd.DataFrame(data)
+    expected_output = pd.DataFrame(
+        data={
+            "A": [1, 2, 3, np.nan, np.nan, np.nan],
+            "B": [np.nan, np.nan, np.nan, 4, 5, 6],
+        },
+        index=range(6),
+    )
+    assert_frame_equal(result, expected_output)
+
+    # Create a DataFrame from the dictionary using the create_dataframe function to observe the fix
+    result = create_dataframe(data)
+    expected_output = pd.DataFrame(
+        {
+            "A": [1, 2, 3],
+            "B": [4, 5, 6],
+        },
+        index=[0, 1, 2],
+    )
+    assert_frame_equal(result, expected_output)
```

Comment:
How do we know we should prioritise the index of A over the index of B?

Reply:
The index of A isn't being prioritised, the indices of both series are reset - I've just pushed a commit to change the index of A to `[1, 2, 3]` to make this clearer

Reply:
Ah of course, we don't even need the index post-transformation so can just reset